### PR TITLE
feat(skymp5-server): reduce damage based on equipped armor

### LINKED ISSUE #337: Reduce damage depending on equipped armor
https://github.com/alekcey0211/red-house-public/blob/e3df9a6c482e8ff2f35fdafba366fefeaf241ad9/functions-lib/src/events/_onHit.ts

UPD:

Formula to implement:

```
armorRating = armorRating1 + armorRating2 + armorRating3 + ... + armorRatingN;
minReceivedDamage = incomingDamage * (1 - 0.01 * fMaxArmorRating)
receivedDamage = std::max(minReceivedDamage, incomingDamage / (armorRating * 0.12 + 1))
```

#### About fMaxArmorRating
Regardless of your displayed armor rating, you can never exceed an 80% physical damage reduction, the armor cap.
fMaxArmorRating is GMST (80.000000 in Skyrim)
https://www.creationkit.com/index.php?title=Category:Settings

### PR DESCRIPTION
**Pull request by nic11**
Closes #337

### CODE REVIEW

**Comment by Pospelove** [comment]
https://en.uesp.net/wiki/Oblivion:The_Complete_Damage_Formula

File: skymp5-server/cpp/lib_espm/espm.h [hunk 1]
```diff
@@ -690,6 +687,7 @@ class WEAP : public RecordHeader
   };
   static_assert(sizeof(WeapData) == 10);
 
+  // XXX: enum class?
```
Comment (Pospelove) [inline skymp5-server/cpp/lib_espm/espm.h:690]: Please remove the comment and convert `enum` to `enum class` then

File: skymp5-server/cpp/lib_espm/espm.h [hunk 2]
```diff
@@ -706,6 +704,7 @@ class WEAP : public RecordHeader
 
   struct DNAM
   {
+    // XXX: enum
     uint8_t animType = 0;
```
Comment (Pospelove) [inline skymp5-server/cpp/lib_espm/espm.h:708]: And here

File: skymp5-server/cpp/server_guest_lib/ActionListener.cpp [hunk 3]
```diff
@@ -10,6 +10,8 @@
 #include "PapyrusObjectReference.h"
 #include "UserMessageOutput.h"
 #include "Utils.h"
+#include "formulas/IDamageFormula.h"
```
Comment (Pospelove) [inline skymp5-server/cpp/server_guest_lib/ActionListener.cpp:13]: Both includes seem unused

**Review by Pospelove — CHANGES_REQUESTED** [review]

File: skymp5-server/cpp/CMakeLists.txt [hunk 4]
```diff
@@ -60,7 +60,7 @@ apply_default_settings(TARGETS geo)
 # server_guest_lib
 #
 
-file(GLOB_RECURSE src "${CMAKE_CURRENT_SOURCE_DIR}/server_guest_lib/*" "${CMAKE_SOURCE_DIR}/.clang-format")
+file(GLOB_RECURSE src "${CMAKE_CURRENT_SOURCE_DIR}/server_guest_lib/**" "${CMAKE_SOURCE_DIR}/.clang-format")
```
Comment (Pospelove) [inline skymp5-server/cpp/CMakeLists.txt:63]: no

Comment (nic11) [reply skymp5-server/cpp/CMakeLists.txt:63]: really no

**Review by Pospelove — CHANGES_REQUESTED** [review]
create TODO's on GitHub and we are good to go testing

File: unit/HitTest.cpp [hunk 5]
```diff
@@ -34,7 +32,8 @@ TEST_CASE("OnHit sends a ChangeValues' packet and damage character by "
 
   REQUIRE(p.Messages().size() == 1);
   auto changeForm = ac.GetChangeForm();
-  REQUIRE(changeForm.healthPercentage == 0.96f);
+  REQUIRE(changeForm.healthPercentage == 0.75f);
+  // XXX: test with other dummy formula? Test with actor with baseHealth > 100?
```
Comment (nic11) [inline unit/HitTest.cpp:36]: Oh and I've forgot about this btw

**Review by Pospelove — APPROVED** [review]

**Pull request merged**